Makefile

```diff
@@ -0,0 +1,173 @@
+# which target is run.
+#
+#     VERSION:
+#         This evaluates the `Version` field of the specfile. Therefore, it will
+#         be set to the latest version number of this repository without any
+#         prefix (just a plain number).
+#
+#     COMMIT:
+#         This evaluates to the latest git commit sha. This will not work if
+#         the source is not a git checkout. Hence, this variable is not
+#         pre-fetched but evaluated at time of use.
+#
+
+VERSION := $(shell (cd "$(SRCDIR)" && grep "^Version:" image-builder-cli.spec | sed 's/[^[:digit:]]*\([[:digit:]]\+\).*/\1/'))
+COMMIT = $(shell (cd "$(SRCDIR)" && git rev-parse HEAD))
+
+#
+# Generic Targets
+#
+# The following is a set of generic targets used across the makefile. The
+# following targets are defined:
+#
+#     help
+#         This target prints all supported targets. It is meant as
+#         documentation of targets we support and might use outside of this
+#         repository.
+#         This is also the default target.
+#
+#     $(BUILDDIR)/
+#     $(BUILDDIR)/%/
+#         This target simply creates the specified directory. It is limited to
+#         the build-dir as a safety measure. Note that this requires you to use
+#         a trailing slash after the directory to not mix it up with regular
+#         files. Lastly, you mostly want this as order-only dependency, since
+#         timestamps on directories do not affect their content.
+#
+
+.PHONY: help
+help:
+	@echo "make [TARGETS...]"
+	@echo
+	@echo "This is the maintenance makefile of image-builder-cli. The following"
+	@echo "targets are available:"
+	@echo
+	@echo "    help:               Print this usage information."
+	@echo "    rpm:                Build the RPM"
+	@echo "    srpm:               Build the source RPM"
+	@echo "    scratch:            Quick scratch build of RPM"
+	@echo "    clean:              Remove all built binaries"
+
+$(BUILDDIR)/:
+	mkdir -p "$@"
+
+$(BUILDDIR)/%/:
+	mkdir -p "$@"
+
+
+#
+# Maintenance Targets
+#
+# The following targets are meant for development and repository maintenance.
+# They are not supported nor is their use recommended in scripts.
+#
+
+.PHONY: build
+build: $(BUILDDIR)/bin/
+	go build -o $<image-builder ./cmd/image-builder/
+
+.PHONY: clean
+clean:
+	rm -rf $(BUILDDIR)/bin/
+	rm -rf $(CURDIR)/rpmbuild
+	rm -rf $(CURDIR)/release_artifacts
+
+#
+# Building packages
+#
+# The following rules build image-builder-cli packages from the current HEAD
+# commit, based on the spec file in this directory.  The resulting packages
+# have the commit hash in their version, so that they don't get overwritten
+# when calling `make rpm` again after switching to another branch.
+#
+# All resulting files (spec files, source rpms, rpms) are written into
+# ./rpmbuild, using rpmbuild's usual directory structure.
+#
+
+RPM_SPECFILE=rpmbuild/SPECS/image-builder-cli.spec
+RPM_TARBALL=rpmbuild/SOURCES/image-builder-cli-$(COMMIT).tar.gz
```

Comment:
Tarball should IMO contain also the release version, not just the commit.

Reply:
I'll align with @mvo5 and @supakeen